internal/terraform/node_resource_abstract_instance.go

```diff
@@ -450,6 +452,18 @@ func (n *NodeAbstractResourceInstance) planDestroy(ctx EvalContext, currentState
 			return plan, diags
 		}
 
+		if resp.Deferred != nil {
+			ctx.Deferrals().ReportResourceInstanceDeferred(n.Addr, resp.Deferred.Reason, &plans.ResourceInstanceChange{
+				Addr: n.Addr,
+				Change: plans.Change{
+					Action: plans.Delete,
+					Before: unmarkedPriorVal,
+					After:  resp.PlannedState,
+				},
```

Comment:
I don't think there's a particular reason we need to save this subset of the planned change as the deferred change. It's be nice if we could save the full change that is computed below.

Reply:
Yeah, that totally makes sense to me 👍